smartcosmos-framework/src/main/java/net/smartcosmos/aspects/SmartCosmosExtensionAspects.java

```diff
@@ -0,0 +1,52 @@
+package net.smartcosmos.aspects;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.AfterThrowing;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.stereotype.Component;
+
+import net.smartcosmos.exceptions.SmartCosmosException;
+
+/**
+ * Aspects to be applied to extension services implemented in the SMART COSMOS Objects architecture.
+ * <p>
+ * To use this aspect you will need to enable aspects with @EnableAspectJAutoProxy in your @Configuration.  Then you need to add
```

Comment:
I wonder if we can tackle this using custom auto-configuration, like _If there is any class annotated with `@SmartCosmosService`, enable aspects._